File: mario.py
# TODO
from cs50 import get_int


def main():
    size = get_size()
    countup = 1
    countdown = size - 1
    for i in range(size):
        print(makespace(countdown), end="")
        print(makeblock(countup), "", makeblock(countup))
        countup += 1
        countdown -= 1


def get_size():
    size = get_int("Height: ")
    if size <= 0 or size >= 9:
        return get_size()
    else:
        return size


def makeblock(num):
    block = "#"
    return block * num


def makespace(num):
    space = " "
    return space * num


main()

File: mario.c
#include <cs50.h>
#include <stdio.h>

int GetSize(void);
void PrintGrid(int n);

int main(void)
{
    int n = GetSize();
    PrintGrid(n);
}

void PrintGrid(int n)
{
    for (int i = 0; i < n; ++i)
    {

        for (int j = n; j > i + 1; --j)
        {
            printf(" ");
        }

        for (int k = 0; k < i + 1; ++k)
        {
            printf("#");
        }

        for (int m = 0; m < i + 2; ++m)
        {

            for (int l = 1; l > m; ++m)
            {
                printf("  ");
            }
            printf("#");
        }

        printf("\n");
    }
}

int GetSize(void)
{

    int n;

    do
    {
        n = get_int("Insert height: ");
    }

    while (n < 1);

    return n;
}
